### PR TITLE
fix: decode filepaths when comparing to the RSST for accurate results

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?

allows filepaths to be URI decoded when comparing to what's in the RSST to prevent special characters from breaking source tracking

### What issues does this PR fix or reference?
@W-13593791@
https://github.com/forcedotcom/cli/issues/2212

### CODE REVIEW

File: src/shared/remoteSourceTrackingService.ts [hunk 1]
```diff
@@ -196,7 +199,7 @@ export class RemoteSourceTrackingService extends ConfigFile<RemoteSourceTracking
     // any item in an aura/LWC bundle needs to represent the top (bundle) level and the file itself
     // so we de-dupe via a set
     Array.from(new Set(elements.flatMap((element) => getMetadataKeyFromFileResponse(element)))).map((metadataKey) => {
-      const revision = revisions[metadataKey];
+      const revision = revisions[decodeURI(metadataKey)];
```
Comment (WillieRuemmele) [inline src/shared/remoteSourceTrackingService.ts:202]: This is the real change, the other changes are just about ordering in the file after removing the eslint rule

File: src/shared/remoteSourceTrackingService.ts [hunk 2]
```diff
@@ -127,7 +124,13 @@ export class RemoteSourceTrackingService extends ConfigFile<RemoteSourceTracking
     }
     return path.isAbsolute(fileToDelete) ? fileToDelete : path.join(process.cwd(), fileToDelete);
   }
-
+  private static convertRevisionToChange(memberKey: string, memberRevision: MemberRevision): RemoteChangeElement {
```
Comment (mshanemc) [inline src/shared/remoteSourceTrackingService.ts:127]: looks like a function to me.

**Review by mshanemc — APPROVED** [review]
no changes required

**Comment by mshanemc** [comment]
QA: used PDR.  (had to remove Profiles from `.forceignore`)

manual create profile named `test (with parens)`
see unencoded name in the `project retrieve preview`
`project retrieve start`
✅ retrieves, stores as an encoded filename (which it does without this change)
`project retrieve preview` no longer sees the change

---
I'm going to merge all the other dependency PRs into this, and also modify the code to check for both encoded and non-encoded key matches.

**Pull request merged**